### PR TITLE
Options grouping

### LINKED ISSUE #132: Add sub-options for each converter and simplify option passing
Now each option is passed as an argument, leading to tremendous number of arguments, i.e. 
`merge_list(files, parsed_args.output, parsed_args.converter, parsed_args.nan, parsed_args.colormap, parsed_args.nscale, ErrorEstimate[parsed_args.error])`

Some options are relevant only to certain converters, i.e. `--colormap`

### PR DESCRIPTION
**Pull request by grzanka**
Fixes #132

### CODE REVIEW

**Comment by grzanka** [comment]
Few examples:

- No command, just help:
```
→ PYTHONPATH=. python3 pymchelper/run.py --help
usage: run.py [-h] [-V] converter ...

positional arguments:
  converter
    ascii        converts to plain ASCII file
    image        converts to PNG images
    plotdata     converts to gnuplot data
    gnuplot      converts to gnuplot script
    tripcube     converts to trip98 data cube

optional arguments:
  -h, --help     show this help message and exit
  -V, --version  show program's version number and exit
```
 - Help for ASCII converter

```
→ PYTHONPATH=. python3 pymchelper/run.py ascii --help
usage: run.py ascii [-h] [--many] [--nan] [--error {none,stderr,stddev}]
                    [-n NSCALE] [-v] [-q] [-V]
                    input [output]

positional arguments:
  input                 input filename, file list or pattern
  output                output filename or directory

optional arguments:
  -h, --help            show this help message and exit
  --many                automatically merge data from various sources
  --nan                 ignore NaN in averaging
  --error {none,stderr,stddev}
                        type of error estimate to add (default: stderr)
  -n NSCALE, --nscale NSCALE
                        Scale with number of primaries N.
  -v, --verbose         Give more output. Option is additive, and can be used
                        up to 3 times
  -q, --quiet           Be silent
  -V, --version         show program's version number and exit
```

- Help for image converter, note new `colormap option`
```
PYTHONPATH=. python3 pymchelper/run.py image --help
usage: run.py image [-h] [--many] [--nan] [--error {none,stderr,stddev}]
                    [-n NSCALE] [-v] [-q] [-V] [--colormap COLORMAP]
                    input [output]

positional arguments:
  input                 input filename, file list or pattern
  output                output filename or directory

optional arguments:
  -h, --help            show this help message and exit
  --many                automatically merge data from various sources
  --nan                 ignore NaN in averaging
  --error {none,stderr,stddev}
                        type of error estimate to add (default: stderr)
  -n NSCALE, --nscale NSCALE
                        Scale with number of primaries N.
  -v, --verbose         Give more output. Option is additive, and can be used
                        up to 3 times
  -q, --quiet           Be silent
  -V, --version         show program's version number and exit
  --colormap COLORMAP   image color map

```

**Comment by grzanka** [comment]
- Standard convertion to ASCII:

```
PYTHONPATH=. python3 pymchelper/run.py ascii data/output/dose20001.bdo d1.txt
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20001.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20001.bdo
INFO:pymchelper.writers.shieldhit:Writing: d1.txt
```

 - Conversion without output filename:
```
→ PYTHONPATH=. python3 pymchelper/run.py ascii data/output/dose20001.bdo       
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20001.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20001.bdo
INFO:pymchelper.writers.shieldhit:Writing: data/output/dose20001.txt
```

- Automatic discovery of scorers, folder conversion (quiet option):
```
PYTHONPATH=. python3 pymchelper/run.py ascii --many "data/output/dose2001*.bdo" out
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20010.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20010.bdo
(...)
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20018.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20018.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20019.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20019.bdo
INFO:pymchelper.writers.shieldhit:Writing: out/dose2.txt
```

**Comment by grzanka** [comment]
- Producing plots:

```
PYTHONPATH=. python3 pymchelper/run.py image --many "data/output/dose2001*.bdo" out
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20010.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20010.bdo
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20011.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20011.bdo
(...)
INFO:pymchelper.readers.shieldhit:Reading header: data/output/dose20019.bdo
INFO:pymchelper.readers.shieldhit:Reading data: data/output/dose20019.bdo
INFO:pymchelper.writers.plots:Writing: out/dose2.txt.png
```

**Comment by nbassler** [comment]
would it be possible to let the ascii version be default behaviour if no command is given? Or perhaps rename ascii -> txt. Many don't know anymore what ascii is. :(

**Comment by nbassler** [comment]
btw: add a few examples of available colourmaps to the help="" or a link to the available maps.

**Comment by nbassler** [comment]
consistency: all help texts should start lowercase

**Comment by grzanka** [comment]
@nbassler:
 - unfortunately there is no easy way to skip the command and provide as first argument the filename to convert.
 - changing ascii to txt sounds reasonable
 - I also agree with lowercase comment

**Comment by grzanka** [comment]
@nbassler ready to be reviewed (and merged later)

File: pymchelper/run.py [hunk 1]
```diff
@@ -27,54 +27,81 @@ def set_logger_level(args):
...
+    subparsers = parser.add_subparsers(dest='command', metavar='converter')
+
+    parser_txt = subparsers.add_parser(Converters.txt.name, help='converts to plain txt file')
+    add_default_options(parser_txt)
+
+    parser_image = subparsers.add_parser(Converters.image.name, help='converts to PNG images')
+    add_default_options(parser_image)
+    parser_image.add_argument("--colormap",
+                              help='image color map, see http://matplotlib.org/users/colormaps.html '
+                                   'for list of possible options',
```
Comment (nbassler) [inline pymchelper/run.py]: :+1:

**Pull request merged**